### PR TITLE
[new release] codept (0.11.0)

### PR DESCRIPTION
**Pull request by Octachron**
This is a new version of codept, an alternative dependency analyzer for OCaml projects.

This new version restores the support for recent compiler (up to 4.13.0). 

It also extends codept to handle every module-level features of OCaml (including abstract modules types). 

Finally, the internal core of codept (the interruptible outliner for OCaml module system) has been entirely switched to the zipper-based engine. 

## Features

  * Support for abstract module types:
	Codept is now aware that `Y` is `X.M.Y` and not an external dependency in

```ocaml
  module F(X:sig module type t module M:t end) = X
  module X = struct
    module M = struct
      module Y = struct end
    end
    module type t = module type of M
  end
  open F(X)
  open M
  open Y
```

  * Support for OCaml 4.09, 4.10, 4.11, 4.12, 4.13

  * Support for split compilation units: the interface and  implementation files do not need to share a filesystem path to be identified as part of the same compilation unit

## Bug fix

 * Nested with constraints `with A.B. ...` triggers now alias dependencies in -no-alias-deps mode
In other words, in this example
```ocaml
module type s = sig module Alias = Deps end
module E = struct end
module type s2 = s with module Deps.Sub = E
```
`Deps` should be counted as an external dependency due to left hand side of the substitution.

* The signature of dune wrapped library can now be imported properly: libraries are now read in the `-no-alias-deps` mode by default.

## Internal

* Switch to the new zipper-based outliner
* Fully typed recursive definition for schema
* Extended testsuite
* More detailled AST for the simplified value level (no more encoding)

### CODE REVIEW

**Comment by Octachron** [comment]
A type error that only appears with `dune -p` and without conjunctive types? That is unexpected.

File: packages/codept/codept.0.11.0/opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+bug-reports: "https://github.com/Octachron/codept/issues"
+license: "gpl-3"
+dev-repo: "git+https://github.com/Octachron/codept.git"
+build: [
+  ["dune" "build" "-p" name]
+]
+run-test: [
+  ["dune" "runtest"]
+]
+depends: ["dune" "menhir" {build} "ocaml" {>="4.03" & < "4.14~"}]
```
Comment (mseri) [inline packages/codept/codept.0.11.0/opam:13]:
```suggestion
depends: [
  "ocaml" {>="4.03" & < "4.14~"}
  "dune" {>= "2.7"}
  "menhir" {build}
  "ocamlfind" {with-test}
]
```
If I looked correctly, the dune-project file requires dune 2.7 and the tests call ocamlfind

**Comment by mseri** [comment]
It can happen when you vendor libraries, which does not seem the case here. I don't know.

File: packages/codept/codept.0.11.0/opam (hunk 1 above)
Comment (Octachron) [reply packages/codept/codept.0.11.0/opam:13]: I am going to relax the dune requirement to 2.5 . Similarly, I will update the test to call `ocamlc -where` rather than ocamlfind to locate `compiler-libs`.

**Comment by Octachron** [comment]
Here, it is the GADT-heavy code for the dynamic type representation that is hitting a pre-4.06 bug ... which is avoided when the `unused-module` warning is enabled (??!). I am thus just going to enable the warning in the release profile.

**Comment by mseri** [comment]
Wow! 🙄

**Comment by mseri** [comment]
Looks good now, thanks!
